### PR TITLE
Build new playground

### PR DESCRIPTION
**Pull request by fifoosid**
- [ ] playground/app not necessary, the whole structure can be directly under playground

### CODE REVIEW

File: packages/main/test/sap/ui/webcomponents/main/qunit/runner.html [hunk 1]
```diff
@@ -56,7 +56,9 @@
 			var params = getUrlQueryParams();
 
 			insertScript(params.testName + ".qunit.js");
-		});
+		};
+
+		loadTests();
```
Comment (vladitasev) [inline packages/main/test/sap/ui/webcomponents/main/qunit/runner.html]: Shouldn't we still wait for document.ready to run the tests? maybe that's why they're failing?

File: packages/playground/app/_config.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Welcome to Jekyll!
```
Comment (vladitasev) [inline packages/playground/app/_config.yml]: Do we need this file?

File: packages/playground/app/Rakefile [hunk 3]
```diff
@@ -0,0 +1 @@
+Dir.glob('lib/tasks/*.rake').each {|r| import r}
```
Comment (vladitasev) [inline packages/playground/app/Rakefile]: Do we need rake?

File: packages/playground/app/_config.yml (hunk 2 above)
Comment (fifoosid) [reply packages/playground/app/_config.yml]: Yes, it is needed

File: packages/playground/app/Rakefile (hunk 3 above)
Comment (fifoosid) [reply packages/playground/app/Rakefile]: Currently we don't use it, but in the future we might start using it

File: packages/playground/app/_includes/head.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    {% if page.description %}
+    <meta name="Description" content="{{ page.description }}">
+    {% endif %}
+  {% endif %}
+
+  <link rel="shortcut icon" href="{{ "/assets/images/favicon.ico" | absolute_url }}" type="image/x-icon">
+
+  <link rel="stylesheet" href="{{ "/assets/css/just-the-docs.css" | absolute_url }}">
+
+  {% if site.ga_tracking != nil %}
```
Comment (adrian-bobev) [inline packages/playground/app/_includes/head.html]: Let's exclude this until we have clear statement from PO

File: packages/playground/app/_layouts/default.html [hunk 5]
```diff
@@ -0,0 +1,70 @@
+<!DOCTYPE html>
+
+<html lang="{{ site.lang | default: "en-US" }}">
+{% include head.html %}
+<body  role="application">
```
Comment (adrian-bobev) [inline packages/playground/app/_layouts/default.html]: you have body tag here but as well in all .sample.html files. Do we have two body tags in the result HTML?

consider removing the body tag from the .sample.html files

File: packages/playground/app/_layouts/empty.html [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<style>
+    .page-header,
+    .side-bar {
+        display: none !important;
```
Comment (adrian-bobev) [inline packages/playground/app/_layouts/empty.html]: !important most probably not needed, since you control all the HTML and CSS

File: packages/playground/app/_layouts/sample.html [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+            c-4.138,3.891-5.582,5.263,15.402,49.425l-5.774,13.691C0,146.097,0,147.838,0,153.33v35.068c0,5.501,0,7.44,46.585,24.127
+            l5.773,13.667c-19.843,44.832-18.51,46.178-14.655,50.032l25.353,24.8l2.522,2.168h2.951c1.525,0,6.092,0,44.685-17.516
+            l14.159,5.758c18.335,45.438,20.218,45.427,25.598,45.427h35.771c5.47,0,7.41,0,24.463-45.589l14.195-5.74
+            c26.014,11,41.253,16.585,45.349,16.585l3.404-0.096l27.479-26.901c3.909-3.945,5.278-5.309-15.589-49.288l5.734-13.702
+            c46.496-17.967,46.496-19.853,46.496-25.221v-35.029C340.268,146.361,340.268,144.434,293.629,127.806z M170.128,228.474
+            c-32.798,0-59.504-26.187-59.504-58.364c0-32.153,26.707-58.315,59.504-58.315c32.78,0,59.43,26.168,59.43,58.315
+            C229.552,202.287,202.902,228.474,170.128,228.474z"/>
+        </g>
+        </g>
+    </g>
```
Comment (adrian-bobev) [inline packages/playground/app/_layouts/sample.html]: Lots of <g> tags. No need of them

File: packages/playground/app/_sass/utilities/_layout.scss [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      .d-#{$media-query}-inline-block { display: inline-block !important; }
+      .d-#{$media-query}-none { display: none !important; }
+
+    }
+  }
+}
+
+// Vertical alignment
+
+.v-align-baseline { vertical-align: baseline !important; }
```
Comment (adrian-bobev) [inline packages/playground/app/_sass/utilities/_layout.scss]: Try to get rid of all !importants here and in other files. No point of having so strong rules

File: packages/playground/app/_sass/vendor/normalize.scss/package.json [hunk 9]
```diff
@@ -0,0 +1,70 @@
+{
```
Comment (adrian-bobev) [inline packages/playground/app/_sass/vendor/normalize.scss/package.json]: If we use a normalize why we need their package.json file?

File: packages/playground/app/assets/css/api.css [hunk 10]
```diff
@@ -1,19 +1,7 @@
...
-	font-family: "72-Regular";
-	src: url(../../../www/home/fonts/72-Regular.woff2) format("woff2"),
-		 url(../../../www/home/fonts/72-Regular.woff) format("woff");
-}
-
 body {
 	font-family: "72-Regular", Arial, Helvetica, sans-serif;
 	box-sizing: border-box;
-	padding: 0.625rem 1.25rem;
+	/* padding: 0.625rem 1.25rem; */
```
Comment (adrian-bobev) [inline packages/playground/app/assets/css/api.css]: just remove it if we do not use it

File: packages/playground/app/assets/js/playground.js [hunk 11]
```diff
@@ -0,0 +1,232 @@
...
+    pageFocus();
+    if (typeof lunr !== 'undefined') {
+        initSearch();
+    }
+    toggleSettings();
+    setTheme();
+});
+
+function toggleSettings() {
+  const settingsButton = document.getElementById("settings-button"),
```
Comment (adrian-bobev) [inline packages/playground/app/assets/js/playground.js]: if we do not run this js through babel, just remove all ES6 from the file. Currently I spotted usage of const/let but maybe there are other things to be changed

File: packages/playground/app/assets/js/playground.js [hunk 12]
```diff
@@ -0,0 +1,232 @@
...
+      this.ref('id');
+      this.field('title', { boost: 20 });
+      this.field('content', { boost: 10 });
+      this.field('url');
+    });
+  
+    // Get the generated search_data.json file so lunr.js can search it locally.
+  
+    sc = document.getElementsByTagName("script");
+    source = '';
```
Comment (adrian-bobev) [inline packages/playground/app/assets/js/playground.js]: mix of ' and " in the file

File: packages/playground/app/assets/js/playground.js [hunk 13]
```diff
@@ -0,0 +1,232 @@
...
+    var index = lunr(function () {
+      this.ref('id');
+      this.field('title', { boost: 20 });
+      this.field('content', { boost: 10 });
+      this.field('url');
+    });
+  
+    // Get the generated search_data.json file so lunr.js can search it locally.
+  
+    sc = document.getElementsByTagName("script");
```
Comment (adrian-bobev) [inline packages/playground/app/assets/js/playground.js]: Not defined variables: sc, source, idx, s, jsPath, jsonPath and I stop looking for more. Just add "use strict" at the top to find them all.

File: packages/playground/app/assets/js/webcomponentsjs/README.md [hunk 14]
```diff
@@ -0,0 +1,229 @@
+[![Build Status](https://travis-ci.org/webcomponents/webcomponentsjs.svg?branch=master)](https://travis-ci.org/webcomponents/webcomponentsjs)
```
Comment (adrian-bobev) [inline packages/playground/app/assets/js/webcomponentsjs/README.md]: no need to have thirdparty readme in our output assets not sure about the license

File: packages/playground/app/assets/js/webcomponentsjs/bundles/webcomponents-ce.js.map [hunk 15]
```diff
@@ -0,0 +1 @@
...cable pending definition for the element, add the\n        // element to the list of elements to be upgraded with that definition.\n        const pendingElements = elementsWithPendingDefinitions.get(localName);\n        if (pendingElements) {\n          pendingElements.push(element);\n        // If there is *any other* applicable definition for the element, add it\n        // to the list of elements with stable definitions that need to be upgraded.\n        } else if (this._internals.localNameToDefinition(localName)) {\n          elementsWithStableDefinitions.push(element);\n        }\n      },\n    });\n\n    // Upgrade elements with 'stable' definitions first.\n    for (let i = 0; i < elementsWithStableDefinitions.length; i++) {\n      this._internals.upgradeElement(elementsWithStableDefinitions[i]);\n    }\n\n    // Upgrade elements with 'pending' definitions in the order they were defined.\n    while (pendingDefinitions.length > 0) {\n      const definition = pendingDefinitions.shift();\n      const localName = definition.localName;\n\n      // Attempt to upgrade all applicable elements.\n      const pendingUpgradableElements = elementsWithPendingDefinitions.get(definition.loca
```
Comment (adrian-bobev) [inline packages/playground/app/assets/js/webcomponentsjs/bundles/webcomponents-ce.js.map]: sourcemaps are also not needed

File: packages/playground/app/assets/js/webcomponentsjs/package.json [hunk 16]
```diff
@@ -0,0 +1,72 @@
+{
```
Comment (adrian-bobev) [inline packages/playground/app/assets/js/webcomponentsjs/package.json]: package.json as well.

Scrolling through the files I am wondering is there a better way to consume the library instead of copy the files directly to the project

File: packages/playground/app/docs/getting-started.html [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+		padding-bottom: 0;
+	}
+
+	@media screen and (max-width: 600px) {
+		.wrapper {
+			padding: 0 1.5rem;
+		}
+	}
+</style>
+<section class="wrapper">
```
Comment (adrian-bobev) [inline packages/playground/app/docs/getting-started.html]: Can we exclude the content of this page as an .md file so I can be easily modified by the doc guys if needed?
It looks like a pretty static content

Just a suggestion, not sure if we need it

File: packages/playground/app/package.json [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+  "license": "MIT",
+  "bugs": "https://github.com/pmarsceill/just-the-docs/issues",
+  "devDependencies": {
+    "stylelint": "^10.0.1",
+    "stylelint-config-primer": "^7.0.0",
+    "stylelint-selector-no-utility": "^4.0.0"
+  },
+  "dependencies": {
+    "@primer/css": "^12.3.1",
+    "npm-run-all": "^4.1.5",
```
Comment (adrian-bobev) [inline packages/playground/app/package.json]: npm-run-all, replace-in-file and rimraf are most probably dev deps.

Not sure about the @primer/css. By scrolling the PR I did not find its usage but if we do not use it in the runtime just move it to the devDeps.

**Comment by fifoosid** [comment]
- [x] the bottom of .md pages is a bit cut, like there's no padding.
- [x] When clicking WC links in the left menu, f.e. table, once the page is realoaded, you don't see what you just clicked on the left. Is it easy to make it scroll down so that the selected component is always in focus?
- [ ] IE: opening the datepicker on the initial page produces a tall dialog with infinite scroll
- [ ] IE: the dialog to change theme/size/rtl does not work properly, popups are initially open, buttons on the bottom bar are not clickable as something overlaps them etc....
- [ ] IE: datepicker icon is missing in the sample page for all datepicker instances
- [x] IE: tabcontainer design is broken - some dots appear
- [x] IE: shellbar title is huge- This is reproducable in master as well
- [x] Overall IE: please check if jekyll does not set CSS rules that interfere with our webcomponents. Since there is no shadow root isolation, each app that wants to run on IE should provide CSS that does not break the internals of WC.

File: packages/playground/package.json [hunk 19]
```diff
@@ -1,50 +1,29 @@
 {
   "name": "@ui5/webcomponents-playground",
   "private": true,
-  "version": "1.0.0-rc.2",
+  "version": "4.0.0",
```
Comment (ilhan007) [inline packages/playground/package.json]: Although not publishes, I think it is better to start with smaller version?

File: packages/main/test/sap/ui/webcomponents/main/qunit/runner.html [hunk 20]
```diff
@@ -5,56 +5,45 @@
...
 	</script>
 
-	<script src="../../../../../../resources/sap/ui/thirdparty/jquery.js"></script>
-	<link rel="stylesheet" type="text/css" media="screen" href="../../../../../../resources/sap/ui/thirdparty/qunit-2.css"/>
-	<script src="../../../../../../resources/sap/ui/thirdparty/qunit-2.js"></script>
-	<script src="../../../../../../resources/sap/ui/qunit/qunit-coverage.js"></script>
-	<script src="../../../../../../resources/sap/ui/thirdparty/sinon.js"></script>
-	<script src="../../../../../../resources/sap/ui/thirdparty/sinon-qunit.js"></script>
+	<!-- Patch all of the dependencies -->
+	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
```
[the remark this answers is not in the document]
Comment (fifoosid) [reply packages/main/test/sap/ui/webcomponents/main/qunit/runner.html]: Done

File: packages/main/test/sap/ui/webcomponents/main/samples/Panel.sample.html [hunk 21]
```diff
@@ -197,11 +151,16 @@ <h3>Panel with Custom Header</h3>
 		</xmp></pre>
 	</section>
 
-	<script>
-		window.prettyPrint();
-	</script>
+<style>
```
[the remark this answers is not in the document]
Comment (fifoosid) [reply packages/main/test/sap/ui/webcomponents/main/samples/Panel.sample.html]: Done

File: packages/playground/app/_layouts/sample.html (hunk 7 above)
Comment (fifoosid) [reply packages/playground/app/_layouts/sample.html]: Done

File: packages/playground/app/_layouts/empty.html (hunk 6 above)
Comment (fifoosid) [reply packages/playground/app/_layouts/empty.html]: Yes, you are right that it is not needed. It is there to make obvious the purpose of this template.

File: packages/playground/app/_sass/vendor/normalize.scss/package.json (hunk 9 above)
Comment (fifoosid) [reply packages/playground/app/_sass/vendor/normalize.scss/package.json]: Done

File: packages/playground/app/assets/css/api.css (hunk 10 above)
Comment (fifoosid) [reply packages/playground/app/assets/css/api.css]: Done

File: packages/playground/app/assets/js/playground.js (hunk 12 above)
Comment (fifoosid) [reply packages/playground/app/assets/js/playground.js]: Done

File: packages/playground/app/assets/js/playground.js (hunk 13 above)
Comment (fifoosid) [reply packages/playground/app/assets/js/playground.js]: This code comes from the theme, but I fixed it.

File: packages/playground/app/assets/js/webcomponentsjs/README.md (hunk 14 above)
Comment (fifoosid) [reply packages/playground/app/assets/js/webcomponentsjs/README.md]: Readme is removed. The license is BSD, so we need to keep the license.md

File: packages/playground/app/assets/js/webcomponentsjs/bundles/webcomponents-ce.js.map (hunk 15 above)
Comment (fifoosid) [reply packages/playground/app/assets/js/webcomponentsjs/bundles/webcomponents-ce.js.map]: Done

File: packages/playground/app/docs/getting-started.html (hunk 17 above)
Comment (fifoosid) [reply packages/playground/app/docs/getting-started.html]: Done

File: packages/main/test/sap/ui/webcomponents/main/qunit/runner.html (hunk 20 above)
[the remark this answers is not in the document]
Comment (pskelin) [reply packages/main/test/sap/ui/webcomponents/main/qunit/runner.html]: still here despite the local version, please remove

File: packages/playground/src/package.json [hunk 22]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (pskelin) [inline packages/playground/src/package.json]: there should be a single package json in `packages/playground` (so this one in `src` should be extracted one level up.

File: packages/playground/src/package.json [hunk 23]
```diff
@@ -0,0 +1,34 @@
...
+    "npm-run-all": "^4.1.5",
+    "replace-in-file": "^4.1.1",
+    "rimraf": "^2.6.3",
+    "nps": "^5.9.8",
+    "cross-env": "^6.0.3"
+  },
+  "scripts": {
+    "test": "stylelint '**/*.scss'",
+    "build": "npm-run-all --parallel prepare:** && bundle exec jekyll build --source ./ --destination ../dist",
+    "build:samples": "cd ../../main && nps build.samples",
```
Comment (pskelin) [inline packages/playground/src/package.json]: instead of building main, we should treat it as a dependency and rely on the root build to build the playground after main.

here we should directly consume the dist folder of main (like main consumes base and does not build it)

**Comment by pskelin** [comment]
When I click on a component from the sidebar, the new component is loaded and then the side bar scrolls to the bottom which is very strange. You can reproduce it with a smaller browser height.

**Comment by pskelin** [comment]
we need to have the same meta tags as the current playground for `description` and `keywords`.

Current:
image

**Comment by MapTo0** [comment]
Decide to fix or not but there are things I don't like:
- placeholder of the search - *Search UI5 Web Components docs* - change the text
- all API documentations (`.md`s) should be part of the playground - not in the github wiki (IMHO)
- it would be nice if we build the docs using `jsdoc`'s templating instead of concatenating but it is fine for now
- **Tutorials** - do we really need to just the `README.md` files of each repo? Can we just link them instead of having nested navigation?
- I am getting some `window.prettyPrint is not a function` errors while clicking here and there
- Single line snippets look a bit ugly
- Popover page looks broken
- I could not generate the API and I had to modify copy script in playground package to `    "copy:samples": "node ./build-scripts/samples-prepare.js \"../main/dist/test-resources/api\"",`
- IMHO: Generation of final samples should be part of the playground not part of the build of each library.
